toolkit/components/uniffi-bindgen-gecko-js/src/templates/UniFFIScaffolding.cpp

```diff
@@ -29,6 +29,9 @@ extern "C" {
 {%- for ci in ci_list %}
 {%- for object in ci.object_definitions() %}
 {%- let pointer_type = ci.pointer_type(object) %}
+// The class exists as a wrapper for a UniFFIPointerType instance for {{ object.nm() }}
```

Comment:
I was unsure about this, but changing this to do what Nika was suggesting would mess up our template flow and we'd have to either lean on askama more there or have more significant changes. If we get push back that is beyond stylistic I'd be okay to do that, but for now, the comment seemed sensible

Reply:
Could something like this work without messing up the flow?

```
constexpr mozilla::uniffi::UniFFIPointerType {{ pointer_type }} = {
  "{{ "{}::{}"|format(ci.namespace(), object.name()) }}"_ns,
  {{ object.ffi_object_free().rust_name() }}
};
```

- I think `constexpr` is the right way to express if it's compatible with `nsCLiteralString`.  If not, I think `static` should work.
- `pointer_type` should probably change it's naming style.  Maybe to `k{ObjectName}PointerType`?

Reply:
My reasoning is that the templates in `ScaffoldingConverter.h` have a template for `template <typename PointerType>`, if we remove the class then we can't differentiate between the different `UniFFIPointerType` using the `typename` - we can find alternative ways to do that (like using askama there) but it seems like it could change the nice structure of the `ScaffoldingConverter`s

Alternatively, there might be a way to use static/compile time values to generate templates, but I haven't looked hard enough

Thoughts?

Reply:
Ahh, I see.  C++ templates allow non-type parameters, so maybe there's a way to make it work, but I think it makes sense to just leave it as-is.

I think I got confused and interpreted "template" to mean the askama template.  What about changing the comment to say something like "it's used as a template parameter in ScaffoldingConverter.h"?